src/enums.h

```diff
@@ -363,4 +363,26 @@ std::array<T, N> vec_to_array(const std::vector<T> &vec)
     return array;
 }
 
+template <typename T>
```

Comment:
Not insisting on a change, but FYI ```std::any_of(c.begin(), c.end(), std::equal_to<type>());``` seems to be exactly what you want.

The supposed benefit of using std::begin(container) is that it allows you to overload std::begin(type) locally (with 'using').  I'm not a big fan of overloading like that, it seems to lead to less-readable code because you have to be extra careful about what code is actually executing all the time, but it can probably be useful sometimes.